c10/util/BFloat16.h

```diff
@@ -14,14 +14,11 @@
 #include <cuda_bf16.h>
 #endif
 
-#if defined(SYCL_EXT_ONEAPI_BFLOAT16_MATH_FUNCTIONS)
```

Comment:
Note that both these headers, for all non-xpu usage, are expected to be header-only and not have any dependency.
I'm not sure what these headers are about, but this might be a very surprising xpu-only behavior looming.

Reply:
Currently, both `CL_SYCL_LANGUAGE_VERSION` and `SYCL_LANGUAGE_VERSION` are defined by the XPU device compiler (SYCL compiler). This is partly a legacy issue from when XPU support was maintained out-of-tree. Moving forward, a cleaner solution would be to guard these headers using USE_XPU to avoid unintended inclusion on other backends. We’ll make sure this doesn’t cause surprises for other backend developers or users. Thanks for pointing it out! I will address this improvement in a follow-up PR. Let's land this one first to unblock the xpu toolchain upgrade.

Reply:
The SYCL compiler supports `BFloat16` like CUDA. The device-specific changes intend to accelerate some BF16 operations.

https://github.com/pytorch/pytorch/blob/b30d276abc0c479a5af7825a43d3adf222f20a16/c10/util/BFloat16.h#L103-L111